src/sentry/integrations/slack/webhooks/event.py

```diff
@@ -79,11 +110,37 @@ def prompt_link(self, slack_request: SlackDMRequest) -> None:
             **SlackPromptLinkMessageBuilder(associate_url).as_payload(),
         }
 
-        client = SlackClient(integration_id=slack_request.integration.id)
-        try:
-            client.post("/chat.postEphemeral", data=payload)
-        except ApiError as e:
-            logger.exception("slack.event.unfurl-error", extra={"error": str(e)})
+        logger_params = {
+            "integration_id": slack_request.integration.id,
+            "team_id": slack_request.team_id,
+            "channel_id": slack_request.channel_id,
+            "user_id": slack_request.user_id,
+            "channel": slack_request.channel_id,
+            **payload,
+        }
+
+        if options.get("slack.event-endpoint-sdk") or slack_request.integration.id in options.get(
+            "slack.event-endpoint-sdk-integration-ids"
+        ):
+            client = SlackSdkClient(integration_id=slack_request.integration.id)
+            try:
+                client.chat_postEphemeral(
+                    channel=slack_request.channel_id,
+                    user=slack_request.user_id,
+                    text=payload["text"],
+                    **SlackPromptLinkMessageBuilder(associate_url).as_payload(),
```

Comment:
nit: this message builder only returns `{"blocks": ...}`, can we just get the blocks out of the payload instead of building them again?